pkg/appgateserver/relay_verifier.go

```diff
@@ -26,7 +26,16 @@ func (app *appGateServer) verifyResponse(
 
 	// Extract the supplier's signature
 	if relayResponse.Meta == nil {
-		return ErrAppGateEmptyRelayResponse
+		payload := relayResponse.GetPayload()
+		payloadBz := make([]byte, payload.Size())
+		if _, err := payload.MarshalTo(payloadBz); err != nil {
+			return ErrAppGateEmptyRelayResponseSignature.Wrapf(
```

Comment:
NIT: Should we be using `Signature` or `Metadata` in the error name?

Reply:
I think I misinterpreted a comment.